haystack/schema.py

```diff
@@ -1610,5 +1621,16 @@ def load(cls, load_dir: Union[str, Path], **read_csv_kwargs):
         # backward compatibility mappings
         for df in node_results.values():
             df.rename(columns={"gold_document_contents": "gold_contexts", "content": "context"}, inplace=True)
+            # convert single document_id to list
+            if "answer" in df.columns and "document_id" in df.columns and not "document_ids" in df.columns:
+                df["document_ids"] = df["document_id"].apply(lambda x: [x], axis=1)
+                df.drop(columns=["document_id"], inplace=True)
+            if (
+                "answer" in df.columns
+                and "custom_document_id" in df.columns
+                and not "custom_document_ids" in df.columns
+            ):
+                df["custom_document_ids"] = df["custom_document_id"].apply(lambda x: [x], axis=1)
+                df.drop(columns=["custom_document_id"], inplace=True)
```

Comment:
Here we enable backward-compatibility with older CSVs. We convert the `document_id` and `custom_document_id` columns for answer CSVs.